packages/@react-stately/toast/src/useToastState.ts

```diff
@@ -10,29 +10,56 @@
  * governing permissions and limitations under the License.
  */
 
-import {ToastStateBase} from '@react-types/toast';
-import {useState} from 'react';
+import {ReactNode, useRef, useState} from 'react';
+import {SpectrumToastProps, ToastState, ToastStateValue} from '@react-types/toast';
 
-
-interface ToastStateProps extends React.HTMLAttributes<HTMLElement> {
-  value?: ToastStateBase[]
+interface ToastStateProps {
+  value?: ToastStateValue[]
 }
 
-export function useToastState(props?: ToastStateProps) {
+const TOAST_TIMEOUT = 5000;
+
+export function useToastState(props?: ToastStateProps): ToastState {
   const [toasts, setToasts] = useState(props && props.value || []);
+  const toastsRef = useRef(toasts);
+  toastsRef.current = toasts;
```

Comment:
Curious about this ref. Did you have to add it cuz `toasts` wasn't up to date in `onRemove`? Just seems strange that `toastRef` is initialized and then set again to be equal to `toasts`

Reply:
`onRemove` called by the toast button worked. There was a closure issue with calling `onRemove` for timeouts. Making this a ref solved that closure issue.